### ISSUE TITLE
Havoc contract balance for external calls

### ISSUE THREAD

**Issue by hajduakos** [SRI-CSL/solidity#115]
When an external call is made, we havoc state variables, but not the balance:
```solidity
contract ExtCallBalance {
    function g() public payable {
    }

    function f() public {
        uint oldBal = address(this).balance; // Old balance
        msg.sender.call(""); // We make an external call with no value

        // But this should fail because callbacks can modify balance
        assert(oldBal == address(this).balance);
    }
}
```

**Comment by hajduakos** [comment]
And since we are havocing state vars, the sum shadow variables should also be havoced

**Comment by hajduakos** [comment]
Fixed in 7ea853b
